src/main/java/softdesign/map/CartesianGridMap.java

```diff
@@ -35,51 +44,33 @@ public Tile getTile(CartesianCoordinate coordinate) {
 	}
 
 	@Override
-	public void markAsCovered(CartesianCoordinate coordinate) {
-		if(getTile(coordinate) != Tile.COVERED) { coveredCount++; }
-
-		setTile(coordinate, Tile.COVERED);
+	public void setTile(CartesianCoordinate coordinate, Tile tile) {
+		try {
```

Comment:
This is proper behaviour. ArrayOutOfBounds just means not passable to us so we return wall. I think this makes a lot of sense

Reply:
Not really willing to check the 4 different overflow conditions upfront. This will not catch any unwanted exceptions and not resort to any unintended behaviour.

Reply:
This is in `#setTile` so I'm not really sure what you mean by "so we return wall"?

It might make a lot of sense for you but it didn't make a lot of sense for me (judging by my first review comment) and the TA won't get it at first glance either.

I think a better solution would be to make a method called `#isEdge` and simply check whether the coordinates exceed arrays `grid[0]` and `grid[0][0]`. Then you can add an explanation to the JavaDocs of that method to explain the why and what.